### DISCUSSION TITLE
Managing Objects across frames

### DISCUSSION THREAD

**Discussion by acode-x** [roboflow/supervision#1635] [Q&A] [answered]
Hi all,

Thank you for building this amazing library! I’ve been exploring it for the past few days and am new to object tracking.
I have a couple of queries related to handling classes and objects in real-life scenarios:

1. `Logging an Error for Stationary Vehicles`: If a vehicle (e.g., a car or truck) remains in the same position for 10 seconds (possibly indicating a breakdown), how can I detect this and log an error message?
2. `Tracking When Specific Classes Cross a LineZone`: How can I determine when a particular class (e.g., a car or truck) crosses a LineZone? I’ve noticed `line_zone.trigger(detections)` returns crossed_in and crossed_out as boolean arrays. Could you explain how to use these arrays to track specific classes and total objects per class crossing the LineZone?

I’d appreciate any insights to help me develop better intuition for working with such scenarios.

ref: https://supervision.roboflow.com/develop/notebooks/count-objects-crossing-the-line/
<img width="320" alt="image" src="https://github.com/user-attachments/assets/f30aee5f-cb0e-4c7f-bf46-fd99fcf69c40">

**Comment by LinasKo** [comment, marked as answer]
Hi @acode-x :wave: 

Thank you for a thorough question!

1. You'll have to apply an object tracker (`sv.ByteTrack`), and record the position of each detection, as well as the time it was first detected or moved: `dict[tracker_id, [detection_center, stationary_since]]`. Check the time when the program started. Then every time you update, if the detection moved by X or more, reset the time. If it didn't move, update the values. This way, you'll know how long each object stood for.

You may need `sv.DetectionsSmoother`, to play around with ByteTrack settings, or to create a custom filtering implementation that avoids jittering or missing detections.

```python
from time import time
`time_now = time()
```

2. When you get the detections from `trigger`, you can filter them like so:

```python
# by class ID
detections = detections[detections.class_id == 0]

# by class name
detections = datections[detections["class_name" == "car"]]
```

**Reply by onuralpszr** [reply]
@LinasKo let's make your answer stronger with this addition :) 

For time based detection you can also use this example as reference : https://github.com/roboflow/supervision/tree/develop/examples/time_in_zone

**Comment by acode-x** [comment]
Thank a lot for this detailed explanation. I am now able to implement both the use cases!
The examples were super useful too. I somehow missed to notice time zone examples in doc.

**Reply by LinasKo** [reply]
Great to hear it!
